src/main/java/de/tum/in/www1/artemis/service/tutorialgroups/TutorialGroupService.java

```diff
@@ -612,4 +617,85 @@ private Optional<User> findStudent(StudentDTO studentDto, String studentCourseGr
+     *
+     * @param course The course for which the tutorial groups should be exported.
+     * @param user   The user for whom the transient properties of the tutorial groups should be set.
+     * @param fields The list of fields to include in the CSV export.
+     * @return A CSV string representing the tutorial groups for the given course and user.
+     * @throws IOException if an I/O error occurs during CSV generation.
+     */
+    public String exportTutorialGroupsToCSV(Course course, User user, List<String> fields) throws IOException {
+        Set<TutorialGroup> tutorialGroups = findAllForCourse(course, user);
+
+        StringWriter out = new StringWriter();
+        CSVFormat csvFormat = CSVFormat.DEFAULT.withHeader(fields.toArray(new String[0]));
+        try (CSVPrinter printer = new CSVPrinter(out, csvFormat)) {
+            for (TutorialGroup tutorialGroup : tutorialGroups) {
+                for (String field : fields) {
+                    switch (field) {
+                        case "ID":
+                            printer.print(tutorialGroup.getId());
+                            break;
+                        case "Title":
+                            printer.print(tutorialGroup.getTitle());
+                            break;
+                        case "Campus":
+                            printer.print(tutorialGroup.getCampus());
+                            break;
+                        case "Language":
+                            printer.print(tutorialGroup.getLanguage());
+                            break;
+                        case "Additional Information":
+                            printer.print(tutorialGroup.getAdditionalInformation());
+                            break;
+                        case "Capacity":
+                            printer.print(tutorialGroup.getCapacity());
+                            break;
+                        case "Is Online":
+                            printer.print(tutorialGroup.getIsOnline());
+                            break;
+                        case "Day of Week":
+                            int dayOfWeek = tutorialGroup.getTutorialGroupSchedule().getDayOfWeek();
+                            printer.print(getDayOfWeekString(dayOfWeek));
+                            break;
+                        case "Start Time":
+                            printer.print(tutorialGroup.getTutorialGroupSchedule().getStartTime());
+                            break;
+                        case "Location":
+                            printer.print(tutorialGroup.getTutorialGroupSchedule().getLocation());
+                            break;
+                        default:
+                            printer.print("");
+                    }
+                }
+                printer.println();
+            }
+        }
+
+        return out.toString();
+    }
+
+    public String getDayOfWeekString(int dayOfWeek) {
```

Comment:
```suggestion
    public String getDayOfWeekString(int dayOfWeek) {
    if (dayOfWeek < 1 || dayOfWeek > 7) {
        throw new IllegalArgumentException("Invalid day of the week: " + dayOfWeek);
    }
    String[] days = {"Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"};
    return days[dayOfWeek - 1];
}
```
This method uses much less lines and is more clear

Reply:
changed 👍